core-plugins/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -0,0 +1,154 @@
+    PluginPropertyUtils.addPluginProp("gcsTargetBucket", "gs://" + gcsTargetBucketName);
+    BeforeActions.scenario.write("GCS target bucket name - " + gcsTargetBucketName);
+  }
+
+  @After(order = 1, value = "@GCS_SINK_TEST")
+  public static void deleteTargetBucketWithFile() {
+    deleteGCSBucket(gcsTargetBucketName);
+    gcsTargetBucketName = StringUtils.EMPTY;
+  }
+
+  private static String createGCSBucket() throws IOException {
+    return StorageClient.createBucket("hdf-e2e-test-" + UUID.randomUUID()).getName();
+  }
+
+  private static String createGCSBucketWithFile(String filePath) throws IOException, URISyntaxException {
+    String bucketName = StorageClient.createBucket("hdf-e2e-test-" + UUID.randomUUID()).getName();
+    StorageClient.uploadObject(bucketName, filePath, filePath);
+    return bucketName;
+  }
+
+  private static void deleteGCSBucket(String bucketName) {
+    try {
+      for (Blob blob : StorageClient.listObjects(bucketName).iterateAll()) {
+        StorageClient.deleteObject(bucketName, blob.getName());
+      }
+      StorageClient.deleteBucket(bucketName);
+      BeforeActions.scenario.write("Deleted GCS Bucket " + bucketName);
+    } catch (StorageException | IOException e) {
+      if (e.getMessage().contains("The specified bucket does not exist")) {
+        BeforeActions.scenario.write("GCS Bucket " + bucketName + " does not exist.");
+      } else {
+        Assert.fail(e.getMessage());
+      }
+    }
+  }
+
+  static String readFile(String path) throws IOException {
+    File file = new File(path);
+    FileInputStream fis = new FileInputStream(file);
+    byte[] data = new byte[(int) file.length()];
+    fis.read(data);
+    fis.close();
+    return new String(data, StandardCharsets.UTF_8);
+  }
+
+  public static void verifyOutputContentAndPartitions() {
+    try {
+      String path = Paths.get(Objects.requireNonNull(TestSetupHooks.class.getResource
+        ("/" + PluginPropertyUtils.pluginProp("joinerOutput"))).getPath()).toString();
+      String expectedOutput = readFile(path);
+      int partitions = 0;
+      StringBuilder sb = new StringBuilder();
+      // The output gcs folder will be like:
+      // hdf-e2e-test-xxxxxx
+      // --2022-06-26-00-27/
+      // ----_SUCCESS
+      // ----part-r-0000
+      // ----part-r-0001
+      // ----part-r-...
+      // The number of part-r-* files should match the expected partitions.
+      for (Blob blob : StorageClient.listObjects(gcsTargetBucketName).iterateAll()) {
+        String name = blob.getName();
+        if (name.contains("part-r")) {
+          partitions++;
+          sb.append(new String((blob.getContent())));
+        }
+      }
+      Assert.assertEquals("Output partition should match",
+                          partitions, Integer.parseInt(PluginPropertyUtils.pluginProp("expectedPartitions")));
+      Assert.assertTrue("Output content should match", Strings.equals(sb.toString(), expectedOutput));
```

Comment:
on a second thought, are we handling the case where data can be shuffled by spark??

Reply:
I think so but I haven't taken a close look at corresponding logic and it seems the output file is ordered according to my recent test after creating 200 partitions, some of which are empty.

Reply:
Added the logic for sorting the output.